gammapy/stats/fit_statistics.py

```diff
@@ -305,10 +312,14 @@ def stat_sum_dataset(cls, dataset):
             npred = npred[mask]
 
             weights = dataset.mask.data[mask].astype("float")
-            return weighted_cash_sum_cython(counts, npred, weights)
+            return get_fit_statistics_compiled()["weighted_cash_sum_compiled"](
```

Comment:
does it slow down the computation?
the compiled fit stat function could also be declared as a class attribute, no?